09_GeometryCreator/main.cpp

```diff
@@ -208,80 +43,55 @@ class GeometryCreatorApp final : public examples::SimpleWindowedApplication
 					return logFail("Couldn't create Command Buffer!");
 			}
 
-			m_winMgr->setWindowSize(m_window.get(), WIN_W, WIN_H);
-			m_surface->recreateSwapchain();
-
-			auto assetManager = make_smart_refctd_ptr<nbl::asset::IAssetManager>(smart_refctd_ptr(system));
-			auto* geometry = assetManager->getGeometryCreator();
-
-			//using Builder = typename CScene::CreateResourcesDirectlyWithDevice::Builder;
-			using Builder = typename CScene::CreateResourcesWithAssetConverter::Builder;
-			auto oneRunCmd = CScene::createCommandBuffer(m_utils->getLogicalDevice(), m_utils->getLogger(), gQueue->getFamilyIndex());
-			Builder builder(m_utils.get(), oneRunCmd.get(), m_logger.get(), geometry);
-
-			// gpu resources
-			if (builder.build())
-			{
-				if (!builder.finalize(resources, gQueue))
-					m_logger->log("Could not finalize resource objects to gpu objects!", ILogger::ELL_ERROR);
-			}
-			else
-				m_logger->log("Could not build resource objects!", ILogger::ELL_ERROR);
+			const uint32_t addtionalBufferOwnershipFamilies[] = {getGraphicsQueue()->getFamilyIndex()};
+			// we want to use the vertex data through UTBs
+			using usage_f = IGPUBuffer::E_USAGE_FLAGS;
+			CAssetConverter::patch_t<asset::ICPUPolygonGeometry> patch = {};
+			patch.positionBufferUsages = usage_f::EUF_UNIFORM_TEXEL_BUFFER_BIT;
+			patch.indexBufferUsages = usage_f::EUF_INDEX_BUFFER_BIT;
+			patch.otherBufferUsages = usage_f::EUF_UNIFORM_TEXEL_BUFFER_BIT;
+			m_scene = CGeometryCreatorScene::create(
+				{
+					.transferQueue = getTransferUpQueue(),
+					.utilities = m_utils.get(),
+					.logger = m_logger.get(),
+					.addtionalBufferOwnershipFamilies = addtionalBufferOwnershipFamilies
+				},patch
+			);
+
+			// TODO: this is plain wrong Arek
+			m_system->mount(make_smart_refctd_ptr<system::CMountDirectoryArchive>(localInputCWD/"../common/include/nbl/examples",smart_refctd_ptr(m_logger),m_system.get()),"nbl/examples");
+			m_system->mount(make_smart_refctd_ptr<system::CMountDirectoryArchive>(localInputCWD/"../common/src/nbl/examples",smart_refctd_ptr(m_logger),m_system.get()),"nbl/examples");
```

Comment:
@AnastaZIuk I think I'll handle this in `MonoAssetManagerAndBuiltinResourceApplication`

Reply:
btw when embed resources is OFF, these should be mounted AND corresponding build dir  which I guess will be just `src/nbl/examples` but somewhere in build dir (if there's for example SPIR-V being made with NSC from stuff in `src/nbl/examples`